src/main/java/cz/cvut/kbss/termit/persistence/dao/VocabularyDao.java

```diff
@@ -197,41 +211,60 @@ public Vocabulary update(Vocabulary entity) {
         }
     }
 
+    /**
+     * Forcefully removes the specified vocabulary.
+     * <p>
+     * This deletes the whole graph of the vocabulary, all terms in the vocabulary's glossary and then removes the vocabulary itself. Extreme caution
+     * should be exercised when using this method. All relevant data, including documents and files, will be dropped.
+     * <p>
+     * Publishes {@link VocabularyRemovalEvent} before the actual removal to allow other services to clean up related resources (e.g., delete the document).
+     * @param entity The vocabulary to delete
+     */
     @ModifiesData
     @Override
     public void remove(Vocabulary entity) {
-        Objects.requireNonNull(entity);
-        try {
-            find(entity.getUri()).ifPresent(elem -> {
-                em.remove(elem);
-                refreshLastModified();
-            });
-        } catch (RuntimeException e) {
-            throw new PersistenceException(e);
-        }
+        eventPublisher.publishEvent(new VocabularyRemovalEvent(this, entity.getUri()));
+        this.removeVocabulary(entity, true);
     }
 
     /**
+     * <p>
+     * Does not publishes the {@link VocabularyRemovalEvent}.<br>
+     * You should use {@link #remove(Vocabulary)} instead.
+     * <p>
      * Forcefully removes the specified vocabulary.
      * <p>
      * This deletes all terms in the vocabulary's glossary and then removes the vocabulary itself. Extreme caution
      * should be exercised when using this method, as it does not check for any references or usage and just drops all
      * the relevant data.
-     *
      * @param entity The vocabulary to delete
+     * @param dropGraph if false,
+     *                  executes {@code  src/main/resources/query/remove/removeGlossaryTerms.ru} removing terms,
+     *                  their relations, model, glossary and vocabulary itself, keeps the document.
+     *                  When true, the whole vocabulary graph is dropped.
      */
-    @ModifiesData
-    public void forceRemove(Vocabulary entity) {
+    public void removeVocabulary(Vocabulary entity, boolean dropGraph) {
```

Comment:
Don't use boolean switches for public APIs. Use two explicitly named methods. They can both call the same internal method with a boolean argument, but the public API will be explicit as to what it does. This will also fix my other issue with this method's name - why is it called `removeVocabulary` and not `remove` (overloading the existing one).